addons/website/static/src/js/widgets/theme.js

```diff
@@ -508,6 +509,12 @@ var ThemeCustomizeDialog = Dialog.extend({
             return self._quickEdit($(inputData));
         }));
 
+        // Handle auto changes
+        var $inputsAuto = $options.find('.o_theme_customize_auto');
+        defs = defs.concat(_.map($inputsAuto, function (inputAuto) {
+            return self._setAuto($(inputAuto), $(inputAuto).siblings('input').prop('checked'));
```

Comment:
Well, my code does not call `_setAuto` for all of them, so basically the situation you expose here is working when I test it. I click on the second one (let's say "Medium Logo"), I have "Medium Logo" activated. Seems to work correctly like for the other two options "Little Logo" or "Big Logo". Or did I miss something in the situation you expose ? But still, indeed it could be nice to use `$enable` / `$disabled` stuff in `_onChange`

Reply:
You have a set of radio inputs which are created for a set of opt. All of these radio inputs should have the same name, so that when you check one all the others are unchecked (this is the browser behavior). Then here, `$options` would be all the radio buttons from this set, and so yes you would could `_setAuto` on all of them. The problem is that all the radio buttons of your set have different names so they are not considered linked. I see you left `xmlid === undefined` as condition for creating unique names, you should change that and adapt the rest. Then there would be only one time where you need to check if an "auto" opt is checked: at its initialization, all the rest would be automatic.

Reply:
@qsm-odoo I've one question remaining.

> Plus, little word of explanation about the switch element (the one alone). If you untick the switch it comes back to default value. So for example on our screenshot, if you untick the switch, it will be Bootstrap font again. It seems logic but I'm clarifying here to make sure this is what we want.

I don't really know how to handle with this anymore. Because in my last push everything else is working but not the behavior explained above. Thanks :)

Reply:
If there is only one "auto" widget in a set, then it is a switch and yes, if you uncheck it, it will go through to `_onChange` -> `_processChange` -> `_setAuto` methods and all should be good ? I'll check your code in a minute

Reply:
Yes, I guess something is wrong into what I've done because I want to use a second argument (boolean) in `_setAuto` function to tell if it is already checked or so and so send 'null' instead of the data-value